helix-term/src/commands.rs

```diff
@@ -5100,3 +5103,93 @@ fn rename_symbol(cx: &mut Context) {
     );
     cx.push_layer(Box::new(prompt));
 }
+
+/// If there is a number at the cursor, increment it by the count.
+fn increment_number(cx: &mut Context) {
+    increment_number_impl(cx, cx.count() as i64);
+}
+
+/// If there is a number at the cursor, decrement it by the count.
+fn decrement_number(cx: &mut Context) {
+    increment_number_impl(cx, -(cx.count() as i64));
+}
+
+/// If there is a number at the cursor, increment it by `amount`.
+fn increment_number_impl(cx: &mut Context, amount: i64) {
+    let (view, doc) = current!(cx.editor);
+    let selection = doc.selection(view.id);
+    let text = doc.text();
+
+    let changes = selection.ranges().iter().filter_map(|range| {
+        if let Some(NumberInfo {
+            range,
+            value: old_value,
+            radix,
+        }) = number_at(text.slice(..), *range)
+        {
+            let new_value = old_value.wrapping_add(amount);
+            let old_text: Cow<str> = text.slice(range.from()..range.to()).into();
+            let prefix = if radix == 10 { "" } else { &old_text[..2] };
+
+            let mut new_text = match radix {
+                2 => format!("{:b}", new_value),
+                8 => format!("{:o}", new_value),
+                10 => format!("{}", new_value.abs()),
+                16 => {
+                    let lower_count = old_text
+                        .chars()
+                        .skip(2)
+                        .filter(char::is_ascii_lowercase)
+                        .count();
+                    let upper_count = old_text
+                        .chars()
+                        .skip(2)
+                        .filter(char::is_ascii_uppercase)
+                        .count();
```

Comment:
minor: You could do this in one iteration by using `fold` and storing two counts `(upper, lower)`